plugin/trino-example-plugin/src/main/java/com/example/ExampleRecordSetProvider.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import com.google.common.collect.Iterables;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorRecordSetProvider;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorSplit;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.InMemoryRecordSet;
+import io.trino.spi.connector.RecordSet;
+import io.trino.spi.type.Type;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+public class ExampleRecordSetProvider
+        implements ConnectorRecordSetProvider
+{
+    private final ExampleConfig config;
+    private final ExampleMetadata metadata;
+
+    @Inject
+    public ExampleRecordSetProvider(ExampleConfig config, ExampleMetadata metadata)
+    {
+        this.config = config;
+        this.metadata = metadata;
+    }
+
+    @Override
+    public RecordSet getRecordSet(
+            ConnectorTransactionHandle connectorTransactionHandle,
+            ConnectorSession connectorSession,
+            ConnectorSplit connectorSplit,
+            ConnectorTableHandle table,
+            List<? extends ColumnHandle> list)
+    {
+        // TODO implement
```

Comment:
Is this TODO for end users? If so, give more of an explanation for what they should do here.

Reply:
I changed this into a proper sentence. I also added a similar TODO in the metadata class. Do you think these are helpful at all?